dropwizard-auth/src/main/java/io/dropwizard/auth/CachingAuthorizer.java

```diff
@@ -45,7 +47,7 @@
     //
     // Field is package-private to be visible for unit tests
     @VisibleForTesting
-    final LoadingCache<ImmutablePair<P, String>, Boolean> cache;
+    final LoadingCache<ImmutableTriple<P, String, UriInfo>, Boolean> cache;
```

Comment:
Could we create a dedicated data class for this triple, for example `AuthorizationContext`?

Reply:
I'll look into it.

Reply:
I introduced an `AuthorizationContext`.
Fair warning, my java is a bit rusty.